promql/lex.go

```diff
@@ -889,3 +889,16 @@ func isDigit(r rune) bool {
 func isAlpha(r rune) bool {
 	return r == '_' || ('a' <= r && r <= 'z') || ('A' <= r && r <= 'Z')
 }
+
+// isLabel reports whether the string can be used as label.
+func isLabel(s string) bool {
```

Comment:
Maybe, but we don't use that for any other identifier parsing in the lexer and I generally tend to avoid regular expressions if they are not necessary.